src/query/sql/src/executor/physical_plan_builder.rs

```diff
@@ -855,8 +862,99 @@ impl PhysicalPlanBuilder {
                 Ok(result)
             }
             RelOperator::Window(w) => {
-                let input = self.build(s_expr.child(0)?).await?;
-                let partition_items = w.partition_by.iter().map(|v| v.index).collect::<Vec<_>>();
+                let input_schema = self.build(s_expr.child(0)?).await?.output_schema()?;
+
+                let mut w = w.clone();
+
+                // Unify the data type for range frame.
+                if w.frame.units.is_range() {
+                    assert!(w.order_by.len() == 1);
+                    let order_by = &mut w.order_by[0].order_by_item.scalar;
+
+                    let mut start = match &mut w.frame.start_bound {
+                        WindowFuncFrameBound::Preceding(scalar)
+                        | WindowFuncFrameBound::Following(scalar) => scalar.as_mut(),
+                        _ => None,
+                    };
+                    let mut end = match &mut w.frame.end_bound {
+                        WindowFuncFrameBound::Preceding(scalar)
+                        | WindowFuncFrameBound::Following(scalar) => scalar.as_mut(),
+                        _ => None,
+                    };
+
+                    let mut common_ty = order_by
```

Comment:
Do we need to remove nullable first and wrap nullable later?

Reply:
Casting to nullable has no effect on Scalar value